app/javascript/pages/admin/Reporting/index.jsx

```diff
@@ -0,0 +1,254 @@
+  const headers = 'Name,Email,Office,Hours\n'
+  const csv = R.reduce((acc, row) => acc + `${row.name},${row.email},${row.office.name},${row.hours}\n`, headers, userData.users)
+  const octetStream = encodeURIComponent(csv)
+  setOctectStream(octetStream)
+}
+
+const ReportingPage = () => {
+  
+  const { t, i18n } = useTranslation()
+  
+  const [ report, setReport ] = useState(null) // null | 'user' | `other report keys`
+  const [ dropDownIsOpen, setDropDownIsOpen ] = useState(false)
+  const [ dropDownTempSelectedItem, setDropDownTempSelectedItem ] = useState(false)
+  const [ octetStream, setOctectStream ] = useState(null)
+
+  const [ dateRange, setDateRange ] = useState(initialRange)
+  const { filters } = useContext(FilterContext)
+
+  const [ getUsers, { loading: userLoading, data: userData } ] = useLazyQuery(ReportingQuery)
+  
+  useEffect(() => {
+    switch (report) {
+      case 'user':
+        userData && setUserOctectStream(userData, setOctectStream)
+        break;
+      // TODO: match on future reports
+      default:
+        break;
+    }
+  }, [userData])
+
+  useEffect(() => {
+    switch (report) {
+      case 'user':
+        getUsers({
+          variables: {
+            after: formatOrDefaultStartDate(dateRange.start),
+            before: formatOrDefaultEndDate(dateRange.end),
+            officeId: R.view(officeFilterValueLens, filters),
+          },
+        })
+        break;
+      // TODO: match on future reports
+      default:
+        break;
+    }
+  }, [ report, dateRange.start, dateRange.end, filters ])
+
+  const changeStartDate = date => setDateRange(R.set(R.lensProp('start'), date))
+  const changeEndDate = date => setDateRange(R.set(R.lensProp('end'), date))
+
+  const reportKeyTitleMap = {
+    user: t('volunteer_portal.admin.tab.reporting.dropdown.users')
+  }
+
+  const handleDropdownStateChange = (changes, stateAndHelpers) => {
+    if (Object.prototype.hasOwnProperty.call(changes, 'isOpen')) {
+      const nextIsOpen =
+        changes.selectedItem === 'general' ||
+        changes.selectedItem === 'organized' ||
+        changes.selectedItem === 'individual' ||
+        changes.isOpen
+      setDropDownIsOpen(nextIsOpen)
+    }
+
+    if (Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
+      const nextSelectedItem = changes.selectedItem
+
+      if (nextSelectedItem === 'general') {
+        switch (dropDownTempSelectedItem) {
+          case 'organized':
+            stateAndHelpers.setHighlightedIndex(1)
+            break
+          case 'individual':
+            stateAndHelpers.setHighlightedIndex(2)
+            break
+        }
+      }
+
+      setDropDownTempSelectedItem(nextSelectedItem)
+    }
+  }
+
+  const handleDropdownOnSelect = item => {
+    if (item !== 'general' && item !== 'individual'  && item !== 'organized') {
+      setReport(item)
+    }
+  }
+
+  const renderMenu = () => (
+    <Menu >
+      {dropDownTempSelectedItem === 'organized' ? (
```

Comment:
This is fine as is, but an alternative pattern that is easier to read and maintain would be to split each case into it's own component and use ramda's `R.cond` ([docs](https://ramdajs.com/docs/#cond)) to switch based on the value of `dropDownTempSelectedItem`.

Typically we only have a component per file so subcomponents usually live in the same folder or a `pages/admin/Reporting/components` folder. 

e.g.
```
      {R.cond([
        [R.equals('organized'), R.always(<OrganisedMenu />)],
        [R.equals('individual'), R.always(<IndividualMenu />)],
        [R.T, R.always(<MainMenu />)],
      ])(dropDownTempSelectedItem)
```

Reply:
Let's try this once we have the other options 😛